python/pypto/runtime/distributed_runner.py

```
@@ -1849,15 +1850,34 @@ def alloc_stacked_tensor(
+                    init=host[i].contiguous(),
+                    worker_id=w,
                 )
+
+            uploaded: dict[int, DeviceTensor] = {}
+            errors: list[Exception] = []
+            with ThreadPoolExecutor(max_workers=len(ids)) as pool:
+                futures = {pool.submit(_upload_shard, i, w): i for i, w in enumerate(ids)}
+                for future, index in futures.items():
+                    try:
+                        uploaded[index] = future.result()
+                    except Exception as exc:  # noqa: PERF203 -- one shard failing must not orphan the rest
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)